ArjunaJTA/cdi/tests/classes/com/hp/mwtests/ts/jta/cdi/transactional/stereotype/extension/NoStereotypeBean.java

```diff
@@ -0,0 +1,27 @@
+package com.hp.mwtests.ts.jta.cdi.transactional.stereotype.extension;
+
+import javax.annotation.Resource;
+import javax.enterprise.inject.Stereotype;
+import javax.transaction.Status;
+import javax.transaction.SystemException;
+import javax.transaction.TransactionManager;
+import javax.transaction.Transactional;
+
+public class NoStereotypeBean {
+    @Resource(mappedName = "java:jboss/TransactionManager")
+    private TransactionManager txnMgr;
+
+    /**
+     * Expecting the {@link AnnotationProcessingExtension} adds the stereotype
+     * over the bean to get the {@link Transactional} defined by {@link Stereotype}
+     * added programmatically in extension.
+     */
+    public void process() throws SystemException {
+        if(Status.STATUS_ACTIVE == txnMgr.getStatus())
+        	throw new RuntimeException("expected status active");
```

Comment:
Testing this by means of forcible exception feels kind of wrong - wouldn't it be "cleaner" to just use a boolean and set it to `true` when there is `STATUS_ACTIVE`?

Reply:
No, exception is needed to be thrown here as then the interceptor starts its handling and I need to test the behaviour of the transactional interceptor in such case. I will add there some comment to say it's for testing purposes.